Source/WebGPU/WebGPU/Texture.mm

```diff
@@ -2417,17 +2432,26 @@ static WGPUExtent3D computeRenderExtent(const WGPUExtent3D& baseSize, uint32_t m
 
     auto slices = NSMakeRange(descriptor->baseArrayLayer, descriptor->arrayLayerCount);
 
-    id<MTLTexture> texture = [m_texture newTextureViewWithPixelFormat:pixelFormat textureType:textureType levels:levels slices:slices];
-    if (!texture)
-        return TextureView::createInvalid(m_device);
+    id<MTLTexture> texture;
+
+    if (m_texture.pixelFormat == pixelFormat && m_texture.textureType == textureType && m_texture.mipmapLevelCount == levels.length && m_texture.arrayLength == slices.length && !levels.location && !slices.location)
+        texture = m_texture;
```

Comment:
How confident are we that `newTextureViewWithPixelFormat` doesn't do this under the hood?

Reply:
`newTextureViewWithPixelFormat` returns `nil` for memoryless textures, that is why this code is needed

Reply:
To limit the scope of this change I can also only invoke this path when `m_texture.storageMode == MTLStorageModeMemoryless`